File: TiendaAPI/Entities/Usuario.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace TiendaAPI.Entities
{
    public class Usuario
    {

        [Key]
        public int idUsuario { get; set; }
        public string nombre { get; set; }
        public string email { get; set; }
        public string foto { get; set; }

        
        [ForeignKey("idTipo")]
        public virtual Tipo Tipo { get; set; }
        public int idTipo { get; set; }
    }
}

File: TiendaApp/src/app/models/usuario.ts
import { Tipo } from './tipo';
export class Usuario {

    constructor(public idUsuario:number,
        public nombre:string,
        public email:string,
        public foto:string,
        public idTipo:number,
        public tipo:Tipo){
            this.idUsuario = idUsuario;
            this.nombre = nombre;
            this.email = email;
            this.foto = foto;
            this.idTipo = idTipo;
            this.tipo = tipo
    }
}
